src/docs/development/platform-integration/add-to-app-android/add-flutter-screen.md

```diff
@@ -0,0 +1,266 @@
+  android:launchMode="singleTop"
+  android:theme="@style/LaunchTheme"
+  android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
+  android:hardwareAccelerated="true"
+  android:windowSoftInputMode="adjustResize"
+  />
+```
+
+Replace `YourFlutterActivityName` with the name of your Flutter `Activity`.
+
+### Step 4: Launch FlutterActivity
+
+With `FlutterActivity` registered in your manifest file, add code to launch `FlutterActivity`
+from whatever point in your app that you'd like. The following example shows `FlutterActivity`
+being launched from an `OnClickListener`.
+
+```java
+myButton.addOnClickListener(new OnClickListener() {
+  @Override
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity.createBuilder().build(currentActivity)
+    );
+  }
+});
+```
+
+The above example assumes that your Dart entrypoint is called `main()`, and your initial
+Flutter route is '/'. If you'd like to change either property, use the `IntentBuilder` to 
+provide your desired properties:
+
+```java
+myButton.addOnClickListener(new OnClickListener() {
+  @Override
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity
+        .createBuilder()
+        .dartEntrypoint("mySpecialEntrypoint")
+        .initialRoute("/my_route")
+        .build(currentActivity)
+      );
+  }
+});
+```
+
+Replace `"mySpecialEntrypoint"` with the name of your desired Dart entrypoint, and replace
+`"/my_route"` with your desired initial route.
+
+### Step 5: (Optional) Use a cached FlutterEngine
+
+Every `FlutterActivity` creates its own `FlutterEngine` by default. Each `FlutterEngine` has
+a non-trivial "warm-up" time. This means that launching a standard `FlutterActivity` comes 
+with a brief delay before your Flutter experience becomes visible. To avoid this delay, you
+can warm-up a `FlutterEngine` before arriving at your `FlutterActivity` and then you can use
+your pre-warmed `FlutterEngine` instead.
+
+To pre-warm a `FlutterEngine`, find a reasonable location in your app to instantiate and hold
+a `FlutterEngine`. For a simplistic app, storing a `FlutterEngine` in the `Application` class
+might be a reasonable option:
+
+```java
+public class MyApplication extends Application {
+  private FlutterEngine flutterEngine;
+
+  @Override
+  public void onCreate() {
+    super.onCreate();
+
+    // Initialize Flutter.
+    FlutterMain.startInitialization(this);
+    FlutterMain.ensureInitializationComplete(this, null);
+
+    // Create and warm-up a FlutterEngine.
+    flutterEngine = new FlutterEngine(this);
+
+    DartExecutor.DartEntrypoint entrypoint = new DartExecutor.DartEntrypoint(
+      getResources().getAssets(),
+      FlutterMain.findAppBundlePath(this),
+      "main"
+    );
+
+    flutterEngine.getDartExecutor().executeDartEntrypoint(entrypoint);
+  }
+
+  @NonNull
+  public FlutterEngine getFlutterEngine() {
+    return flutterEngine;
+  }
+}
+```
+
```

Comment:
Once you call executeDartEntrypoint, your entrypoint Dart code is running (though not rendering). If it invokes runApp, your Flutter widget tree should be prepared for MediaQueryData.size being 0.

Reply:
I'm not sure what you're asking me to do here?

Reply:
The Dart-side entrypoint (the "main" function in this case) is executed as soon as `executeDartEntrypoint` is invoked rather than later when the engine is attached to UI. If your entrypoint calls [runApp()](https://api.flutter.dev/flutter/widgets/runApp.html), the widgets should be prepared to have no [window](https://api.flutter.dev/flutter/dart-ui/window.html) (represented as `MediaQueryData.size == 0`).

Reply:
I still don't know what you're asking me to do. Can you form your comment as a request for a change? I'm aware of how `executeDartEntrypoint` works, but I'm not clear what you'd like me to do about it. Do you want a comment within the code example, or a description before or after the code example, or something else?

If what you're looking for is an explanation of the details you just brought up within the guide, I'm not sure this guide is the right place. Given that we are describing a fullscreen experience, there are numerous tangents that we could choose to take, but those tangents are likely to confuse and distract developers from the task at hand, which is displaying a fullscreen Flutter experience. A race condition, or a single frame property, especially one that is always present regardless of whether you're using `FlutterActivity` or not, is a very specific detail to engrain in this particular guide, in my opinion. Wouldn't it make a lot more sense in a guide about `FlutterEngine`, or a guide about optimizing Flutter execution, or even just a FAQ section?

Reply:
I just meant to insert the paragraph I added :)

Though yes, this page will not be the only article so it can go into a breakout page in the future too.

Reply:
Ok, I added an aside note that explains the implications of executing a Dart entrypoint early. I think it might cause more confusion than clarity, but I guess we can check with devs later.